packages/app/cypress/e2e/runner/support/spec-loader.ts

```diff
@@ -20,6 +20,9 @@ export type LoadSpecOptions = {
   pendingCount?: number | string
   hasPreferredIde?: boolean
   projectName?: 'runner-e2e-specs' | 'session-and-origin-e2e-specs'
+  mode?: 'e2e' | 'component'
```

Comment:
```suggestion
  mode: 'e2e' | 'component'
```

This is not optional, but fully understand if making it non-nullable massively bloats this PR.

Reply:
Absolutely agree that's better, but I did this intentionally to avoid bloating the PR. Ends up touching quite a few places that felt like a unnecessary complication

Reply:
:+1: